pkg/controller/virtualmachineimport/virtualmachineimport_controller.go

```diff
@@ -193,7 +201,19 @@ func (r *ReconcileVirtualMachineImport) Reconcile(request reconcile.Request) (re
 
 	// Define VM spec
 	mapper := provider.CreateMapper()
-	vmSpec := mapper.MapVM(instance.Spec.TargetVMName)
+	template, err := provider.FindTemplate()
```

Comment:
The only thing we need to improve later is , that we shouldn't process the template for each reconcile. It's not neccessary.

Reply:
I think that once the VM is created, there is no need to go through the entire steps.
Perhaps we should store on VMImport CR's annotation/label the identity of the created VM (namespace/name) and skip all of these steps (validation/mapping/templates/...) since we won't re-create the VM.

Reply:
What you mean by identity? The targetVmName?

Reply:
That will work, I'll refactor the loop as part of #51

Reply:
:+1: